### PR TITLE
test model cache management

### PR DESCRIPTION
**Pull request by wanliAlex**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
add new test

* **Other information**:
this test is related to [marqo PR](https://github.com/marqo-ai/marqo/pull/239) and [py-marqo PR](https://github.com/marqo-ai/py-marqo/pull/45)

### CODE REVIEW

File: tests/api_tests/cuda/test_cuda_model_cache_management.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from tests.utilities import classwide_decorate
+
+@classwide_decorate(allow_environments, allowed_configurations=["CUDA_DIND_MARQO_OS"])
+class TestAddDocuments(MarqoTestCase):
+
+    # NOTE: The cuda should already have model loaded in the startup
+    def setUp(self) -> None:
+        self.client = Client(**self.client_settings)
+        self.index_name = "test_index"
+        self.MODEL = "ViT-L/14"
```
Comment (pandu-k) [inline tests/api_tests/cuda/test_cuda_model_cache_management.py]: Change this to the smaller clip model

**Review by pandu-k — CHANGES_REQUESTED** [review]
These tests need to test the format and correctness of the responses, to ensure the API is working properly

File: tests/api_tests/cuda/test_cuda_model_cache_management.py (hunk 1 above)
Comment (wanliAlex) [reply tests/api_tests/cuda/test_cuda_model_cache_management.py]: updated. also add structure test based on the comments from py-marqo.

File: tests/api_tests/test_model_cache_management.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+from tests.utilities import classwide_decorate
+
+
+class TestAddDocuments(MarqoTestCase):
+
+    # NOTE: The cuda should already have model loaded in the startup
+    def setUp(self) -> None:
+        self.client = Client(**self.client_settings)
+        self.index_name = "test_index"
+        self.MODEL = "ViT-L/14"
```
Comment (pandu-k) [inline tests/api_tests/test_model_cache_management.py]: This is still the big model right?

**Review by pandu-k — CHANGES_REQUESTED** [review]

File: tests/api_tests/test_model_cache_management.py (hunk 2 above)
Comment (wanliAlex) [reply tests/api_tests/test_model_cache_management.py]: revised

**Pull request merged**